### PR TITLE
introduce --mate option

### PR DESCRIPTION
**Pull request by robertnurnberg**
Not too sure about what stats to display, so happy to take suggestions. Also some testing would be good.

Examples:
```
Using ./stockfish on matetrack.epd with --mate 3
Engine ID:     Stockfish dev-20240605-5688b188
Total FENs:    44
Found mates:   44
Best mates:    34
Found mates nodes searched: 26214507
Found mates depths reached: 760
```
```
Using ./stockfish on matetrack.epd with --nodes 1000 --mate 0
Engine ID:     Stockfish dev-20240605-5688b188
Total FENs:    6556
Found mates:   48
Best mates:    33
Found mates nodes searched: 51289
Found mates depths reached: 472
```

### CODE REVIEW

File: matecheck.py [hunk 1]
```diff
@@ -242,6 +275,8 @@ def analyze_fens(self, fens):
     print("Total FENs:   ", numfen)
     print("Found mates:  ", mates)
     print("Best mates:   ", bestmates)
+    print("Found mates nodes searched:", totalnodes)
```
Comment (vondele) [inline matecheck.py:278]: is the total most interesting, or the average and maximum?

Comment (robertnurnberg) [reply matecheck.py:278]: I thought avg can be computed from total and `Found mates`.

Comment (robertnurnberg) [reply matecheck.py:278]: Adding max is a good idea, maybe also min. but both will depend on `bm` of course....

**Comment by robertnurnberg** [comment]
New output:
```
> python matecheck.py --mate 3
Loaded 44 FENs, with max(abs(bm)) = 3.

Matetrack started for ./stockfish on matetrack.epd with --mate 3 ...
100%|############################################################################################################| 44/44 [00:15<00:00,  2.82it/s]


Using ./stockfish on matetrack.epd with --mate 3
Engine ID:     Stockfish dev-20240605-5688b188
Total FENs:    44
Found mates:   44
Best mates:    34

Best mate statistics:
abs(bm) = 1 - mates: 4, nodes (min avg max): 4 65 204, depth (min avg max): 1 1 1
abs(bm) = 2 - mates: 7, nodes (min avg max): 209 33933 128334, depth (min avg max): 1 14 41
abs(bm) = 3 - mates: 23, nodes (min avg max): 285 1128383 21564306, depth (min avg max): 6 23 44
```

```
> python matecheck.py --nodes 1000 --mate 0
Loaded 6556 FENs, with max(abs(bm)) = 126.

Matetrack started for ./stockfish on matetrack.epd with --nodes 1000 --mate 0 ...
100%|############################################################################################################| 33/33 [00:27<00:00,  1.22it/s]


Using ./stockfish on matetrack.epd with --nodes 1000 --mate 0
Engine ID:     Stockfish dev-20240605-5688b188
Total FENs:    6556
Found mates:   48
Best mates:    33

Best mate statistics:
abs(bm) = 1 - mates: 4, nodes (min avg max): 4 65 204, depth (min avg max): 1 1 1
abs(bm) = 2 - mates: 3, nodes (min avg max): 209 236 280, depth (min avg max): 1 4 6
abs(bm) = 3 - mates: 5, nodes (min avg max): 285 615 835, depth (min avg max): 6 7 9
abs(bm) = 4 - mates: 2, nodes (min avg max): 473 710 948, depth (min avg max): 12 12 13
abs(bm) = 5 - mates: 3, nodes (min avg max): 340 478 628, depth (min avg max): 6 9 12
abs(bm) = 6 - mates: 1, nodes (min avg max): 120 120 120, depth (min avg max): 11 11 11
abs(bm) = 7 - mates: 11, nodes (min avg max): 160 573 1004, depth (min avg max): 1 11 15
abs(bm) = 8 - mates: 1, nodes (min avg max): 614 614 614, depth (min avg max): 13 13 13
abs(bm) = 10 - mates: 3, nodes (min avg max): 363 7450 17354, depth (min avg max): 1 1 1
```

File: README.md [hunk 2]
```diff
@@ -28,14 +28,17 @@ options:
   --nodes NODES         nodes limit per position, default: 10**6 without other limits, otherwise None (default: None)
   --depth DEPTH         depth limit per position (default: None)
   --time TIME           time limit (in seconds) per position (default: None)
+  --mate MATE           mate limit per position (a value of 0 will use bm #X as the limit) (default: None)
```
Comment (vondele) [inline README.md:31]: maybe add that only those positions where bm <= mate will be run.

**Comment by robertnurnberg** [comment]
Pushed a fix for the bug you experienced:
```
> python matecheck.py --mate 0 --epdFile fast.epd
Loaded 25 FENs, with max(abs(bm)) = 3.

Matetrack started for ./stockfish on fast.epd with --mate 0 ...
100%|############################################################################################################| 25/25 [00:02<00:00,  9.03it/s]


Using ./stockfish on fast.epd with --mate 0
Engine ID:     Stockfish dev-20240605-5688b188
Total FENs:    25
Found mates:   25
Best mates:    25

Best mate statistics:
abs(bm) = 1 - mates: 4, nodes (min avg max): 4 65 204, depth (min avg max): 1 1 1
abs(bm) = 2 - mates: 17, nodes (min avg max): 209 16695 128334, depth (min avg max): 1 19 41
abs(bm) = 3 - mates: 4, nodes (min avg max): 285 15491 45306, depth (min avg max): 7 19 28
```

**Pull request merged**